### ISSUE TITLE
Add support for ad hoc scene imports for Refill themes

### ISSUE THREAD

**Issue by nvkelso** [tangrams/tangram-es#1534]
**In Tangram JS** I can load an import _outside_ of the YAML in the config... 

```javascript
var layer = Tangram.leafletLayer({
        scene: { import: [tg_baseScene, 'theme/brown-orange.yaml'], global: tg_global },
        attribution: '<a href="https://mapzen.com/tangram" target="_blank">Tangram</a>, &copy; OSM contributors'
    });
```

...and I can later modify those ad hoc imports.

- https://github.com/tangrams/refill-style/blob/sensescape/roads-refactor/index.html#L507-L532

``` javascript
scene.load({ import: [ tg_baseScene, import_theme ], global: tg_global });
scene.updateConfig();
```

**But in Tangram ES** it seems like this isn't yet possible. 

@tallytalwar can you add more detail, please?

**Comment by tallytalwar** [comment]
The only time Tangram-es does the `import` process is when the base scene gets loaded and only the `imports` specified in the `base` scene gets imported.

We will need some logic to apply scene imports post scene updates.
Refer code: https://github.com/tangrams/tangram-es/blob/master/core/src/scene/sceneLoader.cpp#L60-L70

It might not be trivial to get this done for tangram-es, but I need to think more on the solution here.

**Comment by bcamper** [comment]
It's not that the `import` is happening "outside" the scene, it's that the entire scene is an ad hoc object (or perhaps more appropriately a string in the case of ES?). @tallytalwar the way this works in JS is to essentially bypass any kind of external load process and just use the provided scene contents. Can we do the same in ES?

**Comment by bcamper** [comment]
FWIW the imports are processed in a similar manner in JS (only done at time of scene load, not on an ad hoc config update by calling `updateConfig()`, etc.).

**Comment by matteblair** [comment]
Correct. Currently Tangram ES can only load scenes from a URL (file, http, etc.). It's not difficult to construct a valid YAML string containing just the desired import list, the trick would just be making that newly-created YAML accessible from a URL somewhere. This is possible, but you would need to write some extra platform-dependent code!

We could add a function to accept a string as a "scene literal" and interpret the string directly as YAML, but that raises some questions about import semantics. If the root scene isn't associated with a URL, then how should import URLs be resolved? It seems like we should only allow absolute URLs for imports in that case. Does that jive with the Tangram JS implementation?

**Comment by bcamper** [comment]
While the usual use cases will be to have absolute import URLs in these cases, there is no restriction on this in Tangram JS. If a relative import is used, it will be resolved to the current document's URL. So there is an implied base path, which generally matches the expectation in a web context (e.g. if you try to load `image.png`, it will load relative to the document).

Is there an analogous resource path we can use in ES? I guess that would be platform-dependent.

**Comment by matteblair** [comment]
There's no analogous URL that behaves consistently across platforms, which is why we've tried to avoid dependence on something like that in Tangram ES.

**Comment by tallytalwar** [comment]
1. Yes we do need to have scene loading done from a yaml string directly.
2. With respect to resolving the issue on what will be the base url for which the resources will be resolved against, take this for an example:
```
loadScene("{import: [ baseScene.yaml, specific-theme.yaml ]}");
```
We can probably use the first yaml in the "direct" import list to construct the base url using the resourcePath(platform specific).

**Comment by bcamper** [comment]
Adjacent imports shouldn't have any effect on each other's base paths.
Either the top level scene has a base path, or it doesn't (in which case I
think requiring absolute URLs for imports in "ad hoc" scenes is ok).

**Comment by matteblair** [comment]
Per the suggestion of @bcamper we can make specifying a base URL a part of this "scene literal" interface, something like:
```cpp
void loadSceneFromSourceString(std::string baseUrl, std::string sceneSource);
```
This removes any ambiguity about how URLs in the scene source will be resolved.

**Comment by tallytalwar** [comment]
Addressed in #1577
